tests/unit/introspection.tcl

```diff
@@ -273,24 +273,48 @@ start_server {tags {"introspection"}} {
 
     test {CONFIG SET rollback on error} {
         # backup maxmemory config
-        set backup [lindex [r config get maxmemory] 1]
+        set mm_backup [lindex [r config get maxmemory] 1]
+        set mmc_backup [lindex [r config get maxmemory-clients] 1]
         # Set some value to maxmemory
         assert_equal [r config set maxmemory 10000002] "OK"
         # Set another value to maxmeory together with another invalid config
-        catch {r config set maxmemory 10000001 maxmemory-clients invalid} e
-        assert_match "ERR Invalid arguments*" $e
-        # Make sure we reverted back to the previous maxmemory
-        r config set maxmemory $backup
+        assert_error "ERR Invalid arguments*" {r config set maxmemory 10000001 maxmemory-clients invalid}
+        # Validate we rolled back to original values
+        assert_equal [lindex [r config get maxmemory] 1] 10000002
+        assert_equal [lindex [r config get maxmemory-clients] 1] $mmc_backup
+        # Make sure we revert back to the previous maxmemory
+        r config set maxmemory $mm_backup
+    }
+
+    test {CONFIG SET rollback on apply error} {
+        # This test tries to configure a used port number in redis. This is expected
+        # to pass the `CONFIG SET` validity checking implementation but fail on 
+        # actual "apply" of the setting. This will validate that after an "apply"
+        # failure we rollback to the previous values.
+        proc dummy_accept {chan addr port} {}
+        
+        set port_backup [lindex [r config get port] 1]
+        set used_port [expr ($port_backup+1)%65536]
+
+        # Run a dummy server on used_port so we know we can't configure redis to 
+        # use it. It's ok for this to fail because that means used_port is invalid 
+        # anyway
+        catch {socket -server dummy_accept $used_port}
+        # Try to listen on the used port
+        assert_error "ERR Invalid arguments*" {r config set port $used_port}
```

Comment:
i'd like to add another config before the port (like you did with maxmemory above), so we can test the first one was reverted too.
ideally, it would be one with an apply function, which we can test too (like oom-score-adj, but preferably not one that's platform specific)

Reply:
I couldn't find any other good apply function other than port (except tls, but then the test becomes tls specific). I did however add more configs before/after `port` so I think we can resolve this.

Reply:
what about `oom-score-adj-values`?
in order to properly validate that, the tcl code will need to have some linux specific portion, but maybe it's not that bad.

Reply:
I can do this, but then the test will run only on linux and I don't like it that much. If you're sure it's better to have the linux specific test here than what's currently there then let me know. My default is not to change it.

Reply:
i didn't mean that the test will run only on linux, just that the part of verifying the rollback was done correctly (looking into `/proc` will be linux specific.

p.s. i don't mind such a test being linux specific since:
1. linux is our main target platform.
2. if we test that logic on linux, there's no reason to assume it'll somehow get broken on another platform.

anyway, if you don't think it worth the effort you can drop it.
it's just that this rollback mechanism is complicated, and has multiple steps, so i feel it should be tested heavily.

Reply:
Did it and it indeed found some bugs. Fixed them. Please review.